internal/client/client.go

```diff
@@ -545,7 +547,10 @@ func (a *APIClient) SwitchContext(name string) error {
 	if err := a.config.SwitchContext(name); err != nil {
 		return err
 	}
-	if err := a.invalidateCache(); err != nil {
+	err := a.invalidateCache()
```

Comment:
@driv Good catch! I think we don't need to declare a sentinel error here since the only failure in invalidateCache is no connection.
However, I do think we should move invalidateCache once we've validated we have a good connection for the new context. 
Does this make sense?

Reply:
I'm not sure I understand, you mean invalidateCache should be only called when there is a good connection?

That would mean doing the same check `a.getConnOK()` that happens in CachedDiscovery, right?

Reply:
Updated.